packages/cozy-client/src/CozyClient.js

```diff
@@ -239,6 +239,36 @@ class CozyClient {
     })
   }
 
+  /**
+   * In cozy app context, CozyClient can be instantiated from data injected by the stack
+   *
+   * @param  {string}   selector - Options
+   * @param  {object}   options  - CozyClient constructor options
+   * @returns {Object} - CozyClient instance
+   */
+  static fromDOM(selector = '[role=application]', options = {}) {
+    const root = document.querySelector(selector)
+    if (!root || !root.dataset) {
+      throw new Error(
+        `Found no data in ${selector} to instantiate cozyClient`
+      )
+    }
+
+    const {cozyDomain, cozyToken} = root.dataset
```

Comment:
We should supports the both way the stack injects the content : 
- in the dataset 
- but also in JSON format in data-cozy 

```suggestion
const data = root.dataset.cozy
      ? JSON.parse(root.dataset.cozy)
      : { ...root.dataset }
const {cozyDomain, cozyToken} = data
```